test/unit/renderer/dataframe.test.js

```diff
@@ -248,6 +248,141 @@ describe('src/renderer/Dataframe', () => {
             expect(pointInTriangle({ x: 0, y: 1 }, { x: 0, y: 1 }, { x: 0, y: -1 }, { x: 0, y: -1 })).toBe(false);
         });
     });
+
+    describe('.getCentroid', () => {
+        describe('when dataframe is point type', () => {
+            const dataframe = new Dataframe({
+                center: { x: 0, y: 0 },
+                scale: 1,
+                geom: new Float32Array([
+                    0, 0,
+                    0, 0,
+                    0, 0,
+
+                    1, 1,
+                    1, 1,
+                    1, 1
+                ]),
+                properties: {
+                    id: [1]
+                },
+                type: 'point',
+                size: 1,
+                active: true,
+                metadata: new Metadata({
+                    properties: {
+                        id: {
+                            type: 'number'
+                        }
+                    },
+                    idProperty: 'id'
+                })
+            });
+            const feature1 = { id: 1 };
+            dataframe.renderer = { _zoom: 1, gl: { canvas: { width: 1024 * window.devicePixelRatio, height: 1024 * window.devicePixelRatio } } };
+            dataframe.matrix = m;
+
+            it('should return the centroid for the required feature', () => {
+                const [x1, y1] = dataframe.getCentroid(feature1.id);
+                expect(x1).toBeCloseTo(180.00, 2);
+                expect(y1).toBeCloseTo(85.05, 2);
+            });
+        });
+
+        xdescribe('when dataframe is line type', () => {
```

Comment:
why `x`?

Reply:
They are currently not working as they are configured. I should do more work to set them up

Reply:
I have updated / improved the examples